qa: failure in p2p_segwit

Saw this the first time and keeping track of it by filing an issue.

```
File "p2p-segwit.py", line 1006, in test_block_relay
    assert_equal(wit_block.serialize(True), hex_str_to_bytes(rpc_block))
```
Commit: 927a1d7d088e52aa079682e1d4f514222c0a2069
Job: https://travis-ci.org/bitcoin/bitcoin/jobs/302552842

The byte strings copied from the log:

```
AssertionError: not(
b'\x04\x00\x00\x00\xfb\xc8t_\xec\xa0\xd3\x17}X\x8c$\xe1\x857\t\x14\xbd\xe0\x97B\xed\xb9R\xb5\x9f\xfb\xd5n\xbc\x1c!\xe9\x80\xa1M7\xd7\xba!\xd5\x03\x9bd\xa9s\xb3\xb9\xf1\xd2b\x8d\xa1\xf4\xa81\xa6cP\xb5\x02:\xc9\x94\xf4t\x0cZ\xff\xff\x7f \x00\x00\x00\x00\x01\x01\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff\xff\xff\x03\x02\xc3\x01\xff\xff\xff\xff\x02@\xbe@%\x00\x00\x00\x00\x01Q\x00\x00\x00\x00\x00\x00\x00\x00&j$\xaa!\xa9\xed\xe2\xf6\x1c?q\xd1\xde\xfd?\xa9\x99\xdf\xa3iSu\\i\x06\x89y\x99b\xb4\x8b\xeb\xd86\x97N\x8c\xf9\x00\x00\x00\x00' == 
b'\x04\x00\x00\x00\xfb\xc8t_\xec\xa0\xd3\x17}X\x8c$\xe1\x857\t\x14\xbd\xe0\x97B\xed\xb9R\xb5\x9f\xfb\xd5n\xbc\x1c!\xe9\x80\xa1M7\xd7\xba!\xd5\x03\x9bd\xa9s\xb3\xb9\xf1\xd2b\x8d\xa1\xf4\xa81\xa6cP\xb5\x02:\xc9\x94\xf4t\x0cZ\xff\xff\x7f \x00\x00\x00\x00\x01\x01\x00\x00\x00\x00\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff\xff\xff\x03\x02\xc3\x01\xff\xff\xff\xff\x02@\xbe@%\x00\x00\x00\x00\x01Q\x00\x00\x00\x00\x00\x00\x00\x00&j$\xaa!\xa9\xed\xe2\xf6\x1c?q\xd1\xde\xfd?\xa9\x99\xdf\xa3iSu\\i\x06\x89y\x99b\xb4\x8b\xeb\xd86\x97N\x8c\xf9\x01 \x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')

Another one: 

https://ci.appveyor.com/project/DrahtBot/bitcoin/builds/31079510#L4118

```
                                   File "C:\projects\bitcoin/test/functional/p2p_segwit.py", line 422, in test_block_relay
                                       assert_equal(wit_block.serialize(), hex_str_to_bytes(rpc_block))
                                     File "C:\projects\bitcoin\test\functional\test_framework\util.py", line 46, in assert_equal
                                       raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
                                   AssertionError: not(b"\x04\x00\x00\x00\xeb\x97\xc4\xa9\x0f2FS\xef\xde5\xe8N\x83u/\xaa?\xec\x88\xb9\xf2\x0eh\xea\x9bu\xb6\xa7dHx\xfe\xae\xb1K\xa5,\x94\x96:\xdcP b\xc1\xd4N\xfe\xe8a\xd8\x14\x10\x89\xd1'\xc4\x10?\xb8\xf3\xa1\xb1\xfd\xedV^\xff\xff\x7f \x00\x00\x00\x00\x01\x01\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff\xff\xff\x03\x02\x8b\x00\xff\xff\xff\xff\x02\x00\xf2\x05*\x01\x00\x00\x00\x01Q\x00\x00\x00\x00\x00\x00\x00\x00&j$\xaa!\xa9\xed\xe2\xf6\x1c?q\xd1\xde\xfd?\xa9\x99\xdf\xa3iSu\\i\x06\x89y\x99b\xb4\x8b\xeb\xd86\x97N\x8c\xf9\x00\x00\x00\x00" == b"\x04\x00\x00\x00\xeb\x97\xc4\xa9\x0f2FS\xef\xde5\xe8N\x83u/\xaa?\xec\x88\xb9\xf2\x0eh\xea\x9bu\xb6\xa7dHx\xfe\xae\xb1K\xa5,\x94\x96:\xdcP b\xc1\xd4N\xfe\xe8a\xd8\x14\x10\x89\xd1'\xc4\x10?\xb8\xf3\xa1\xb1\xfd\xedV^\xff\xff\x7f \x00\x00\x00\x00\x01\x01\x00\x00\x00\x00\x01\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff\xff\xff\x03\x02\x8b\x00\xff\xff\xff\xff\x02\x00\xf2\x05*\x01\x00\x00\x00\x01Q\x00\x00\x00\x00\x00\x00\x00\x00&j$\xaa!\xa9\xed\xe2\xf6\x1c?q\xd1\xde\xfd?\xa9\x99\xdf\xa3iSu\\i\x06\x89y\x99b\xb4\x8b\xeb\xd86\x97N\x8c\xf9\x01 \x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00")

--- a/test/functional/p2p_segwit.py
+++ b/test/functional/p2p_segwit.py
@@ -147,6 +147,11 @@ class TestP2PConn(P2PInterface):
         super().__init__()
         self.getdataset = set()
 
+    # Avoid sending out msg_getdata in the mininode thread as a reply to invs.
+    # They are not needed and would only lead to races because we send msg_getdata out in the test thread
+    def on_inv(self, message):
+        pass
+
     def on_getdata(self, message):
         for inv in message.inv:
             self.getdataset.add(inv.hash)

